### ISSUE TITLE
Invalid code on device flow user code page throws

### ISSUE THREAD

**Issue by brockallen** [IdentityServer/IdentityServer4#2778]
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at IdentityServer4.Services.DefaultDeviceFlowInteractionService.GetAuthorizationContextAsync(String userCode) in C:\ballen\github\identity\IdSvr4\IdentityServer4\src\Services\Default\DefaultDeviceFlowInteractionService.cs:line 36
   at IdentityServer4.Quickstart.UI.Device.DeviceController.BuildViewModelAsync(String userCode, DeviceAuthorizationInputModel model) in C:\ballen\github\identity\IdSvr4\IdentityServer4\host\Quickstart\Device\DeviceController.cs:line 145
   at IdentityServer4.Quickstart.UI.Device.DeviceController.UserCodeCapture(String userCode) in C:\ballen\github\identity\IdSvr4\IdentityServer4\host\Quickstart\Device\DeviceController.cs:line 59
   at Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeNextActionFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeInnerFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeNextResourceFilter()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeFilterPipelineAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync()
   at Microsoft.AspNetCore.Builder.RouterMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware.Invoke(HttpContext context)
   at IdentityServer4.Hosting.IdentityServerMiddleware.Invoke(HttpContext context, IEndpointRouter router, IUserSession session, IEventService events) in C:\ballen\github\identity\IdSvr4\IdentityServer4\src\Hosting\IdentityServerMiddleware.cs:line 72
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Cors.Infrastructure.CorsMiddleware.Invoke(HttpContext context)
   at IdentityServer4.Hosting.BaseUrlMiddleware.Invoke(HttpContext context) in C:\ballen\github\identity\IdSvr4\IdentityServer4\src\Hosting\BaseUrlMiddleware.cs:line 36
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

**Comment by scottbrady91** [comment]
Addressed in #2798

**Comment by leastprivilege** [comment]
done
